git_machete/client.py

```diff
@@ -747,6 +747,8 @@ def traverse(
                         self.__down_branches[upstream])
                     if branch in self.__annotations:
                         del self.__annotations[branch]
+                    if branch in self.managed_branches:
```

Comment:
When trying to iterate over `managed_branches` in `status` function` i encounter bug where self.managed_branches were not updated properly ->`slide_out` function removes slid out branches from the `self.managed_branches` but `traverse` did not and i don't see a reason why it should not, @PawelLipski wdyt?

Reply:
Pls investigate this particular change on a separate PR as discussed